### PR TITLE
Add links for Plug callbacks

### PR DESCRIPTION
**Pull request by sabiwara**
Make references to Plug callbacks valid links for easier navigation.

Screenshot from 2021-02-16 16-11-00

### CODE REVIEW

**Review by eksperimental — COMMENTED** [review]
Hi @sabiwara 
I think it is better to use reference links since they are links that are used repetedly.

Please read this discussion: https://github.com/phoenixframework/phoenix/pull/4202#pullrequestreview-588497696

**Comment by sabiwara** [comment]
> Hi @sabiwara
> I think it is better to use reference links since they are links that are used repetedly.
> 
> Please read this discussion: [#4202 (review)](https://github.com/phoenixframework/phoenix/pull/4202#pullrequestreview-588497696)

Hi @eksperimental , thank you for the great suggestion, it makes the markdown much cleaner :+1:

File: guides/plug.md [hunk 1]
```diff
@@ -301,3 +301,6 @@ To make this all work, we converted the nested blocks of code and used `halt(con
 At the end of the day, by replacing the nested blocks of code with a flattened series of plug transformations, we are able to achieve the same functionality in a much more composable, clear, and reusable way.
 
 To learn more about plugs, see the documentation for the [Plug project](https://hexdocs.pm/plug), which provides many built-in plugs and functionalities.
+
```
Comment (eksperimental) [inline guides/plug.md:304]: Nitpicking, I would add two empty lines since it is the end of the page and what follows is just reference links

Comment (sabiwara) [reply guides/plug.md:304]: Good point :+1:

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**